src/redux/books/books.js

```diff
@@ -1,49 +1,64 @@
-const ADD_BOOK = 'ADD_BOOK';
-const REMOVE_BOOK = 'REMOVE_BOOK';
-const books = [
-  {
-    id: '1',
-    name: 'The Hunger Games',
-    categorie: 'Action',
-    author: 'Suzanne Collins',
-    completionStatus: 75,
-    currentChapter: 'Chapter 17',
-  },
-  {
-    id: '2',
-    name: 'Dune',
-    categorie: 'Science Fiction',
-    author: 'Frank Herbert',
-    completionStatus: 50,
-    currentChapter: 'Chapter 3: "A Lesson Learned"',
-  },
-  {
-    id: '3',
-    name: 'Capital in the Twenty-First Century',
-    categorie: 'Economy',
-    author: 'Suzanne Collins',
-    completionStatus: 0,
-    currentChapter: 'Introduction',
-  },
-];
-
-export const addBook = (book) => ({
-  type: ADD_BOOK,
-  payload: book,
-});
-
-export const removeBook = (id) => ({
-  type: REMOVE_BOOK,
-  id,
-});
-
-export const bookReducer = (state = books, action) => {
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+const ADD_BOOK = 'bookstore/books/ADD_BOOK';
+const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';
+const GET_BOOK = 'bookstore/books/GET_BOOK';
+
+export const bookReducer = (state = [], action) => {
   switch (action.type) {
-    case ADD_BOOK:
+    case `${GET_BOOK}/fulfilled`:
+      return [...state, ...action.payload];
+    case `${ADD_BOOK}/fulfilled`:
       return [...state, action.payload];
-    case REMOVE_BOOK:
-      return state.filter((book) => book.id !== action.id);
+    case `${REMOVE_BOOK}/fulfilled`:
+      return state.filter((book) => book.item_id !== action.payload);
     default:
       return state;
   }
 };
+
+const getAllBooks = async () => {
+  const result = await fetch(process.env.REACT_APP_BASE_URL);
+  return result.json();
+};
+
+const createBook = async (book) => {
+  const result = await fetch(process.env.REACT_APP_BASE_URL, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(book),
+  });
+
+  return result;
+};
+
```

Comment:
- [ ] I seem to be having a similar challenge with the previous reviewer @danmainah . When I create a book, it gets created, however, it seems not to be persisting and sent to the API endpoint thus when I refresh the page it disappears. Kindly look into why this is happening.

**NOTE:** I have cloned it locally and the above screencast is from my local server and not the online version.

Reply:
@Nemwel-Boniface did you add your own api link? because I'm not uploading my own .env file. And it's working on my side. Please try to add comment next time before making me loose another request, thanks

Reply:
https://www.loom.com/share/f0dc76845a46445ea6c14a19360b97f6